src/main/java/com/google/devtools/build/lib/analysis/actions/CustomCommandLine.java

```diff
@@ -1297,32 +1249,33 @@ private ImmutableList<String> argumentsInternal(@Nullable ArtifactExpander artif
         arg = substituteTreeFileArtifactArgvFragment((TreeFileArtifactArgvFragment) arg);
       }
       if (arg instanceof NestedSet) {
-        evalSimpleVectorArg(((NestedSet<?>) arg).toList(), builder);
+        evalSimpleVectorArg(((NestedSet<?>) arg).toList(), builder, pathMapper);
       } else if (arg instanceof Iterable) {
-        evalSimpleVectorArg((Iterable<?>) arg, builder);
+        evalSimpleVectorArg((Iterable<?>) arg, builder, pathMapper);
       } else if (arg instanceof ArgvFragment) {
         if (artifactExpander != null && arg instanceof TreeArtifactExpansionArgvFragment) {
           TreeArtifactExpansionArgvFragment expansionArg = (TreeArtifactExpansionArgvFragment) arg;
           expansionArg.eval(builder, artifactExpander);
         } else {
-          i = ((ArgvFragment) arg).eval(arguments, i, builder, getPathStripper());
+          i = ((ArgvFragment) arg).eval(arguments, i, builder, pathMapper);
         }
       } else if (arg instanceof ActionInput) {
-        builder.add(getPathStripper().getMappedExecPathString((ActionInput) arg));
+        builder.add(pathMapper.getMappedExecPathString((ActionInput) arg));
       } else if (arg instanceof PathFragment) {
-        builder.add(getPathStripper().map((PathFragment) arg).getPathString());
+        builder.add(pathMapper.map((PathFragment) arg).getPathString());
       } else {
         builder.add(CommandLineItem.expandToCommandLine(arg));
       }
     }
     return builder.build();
   }
 
-  private void evalSimpleVectorArg(Iterable<?> arg, ImmutableList.Builder<String> builder) {
+  private void evalSimpleVectorArg(Iterable<?> arg, ImmutableList.Builder<String> builder,
+      PathMapper pathMapper) {
     for (Object value : arg) {
       builder.add(
           value instanceof ActionInput
```

Comment:
Any idea why the check is specifically check for only ActionInput? I'm seeing `CustomCommandLines.arguments` return a list of arguments and the arguments could be different type including `Iterable` which is then handled in this method. I observed that the `Iterable` could contains `String` that needs to be mapped as well. This is a regression with this PR. I'm looking into where this was previously handled.

The affected mnemonic is `Turbine` and it's not a Starlark action. M
![Screenshot 2023-09-21 at 4 23 04 PM](https://github.com/bazelbuild/bazel/assets/91614797/3f0bb9cd-8cb0-4401-9d1d-af7d96c4b1b1)

Reply:
Could you share the Turbine argument that isn't stripped appropriately? The screenshot only shows `--output`, which I assume is stripped. Is it `--javacopt` or something else?

Reply:
I think it is javacopts under the format "-XepOpt:foo:bar=blaze-out/k8-opt-exec-ST-50b1e42ff8f2/genfiles/baz"

Reply:
Thanks, I will add a test for this in Bazel!

Reply:
This regressed in https://github.com/bazelbuild/bazel/commit/17468af6591564a752fa8eaeb5a9032569bda4c8, which is used by `JavaCompileActionBuilder`. I pushed two new commits: the first adds support for `CustomCommandLine.Builder#addObject`, the second adds a test (Java analysis test, not aquery-based).

Reply:
very nice! I was able to rerun the failed test and some more sample testings with the new commits and it looks good so far. I'll kick off a bigger testing round tonight and I'll keep you updated. Thank you so much!